### PR TITLE
Private Events

### PR DESCRIPTION
**Pull request by dasileker**
## Private Events

Created as part of The Odin Project's [Rails Project: Associations](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations?ref=lnav), this app allows users to organize private event networks through invitations.

Currently, attendees must be added to events via the command line. Invitations will be added later.

# Built With

* Passion purple_heart for code.
* `ruby` v2.7.2
* `rails` v6.1
* `vscode` with _Ruby_ and _Rubocop_ extensions
* Linux

### CODE REVIEW

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+      redirect_to @user
+    else
+      render 'new'
+    end
+  end
+
+  def show
+    @user = User.find(params[:id])
+    @organized_events = @user.organized_events
+    redirect_to root_url and return unless @user == current_user
```
Comment (SpaYco) [inline app/controllers/users_controller.rb]: * [x] list events previoulsy attendented by user **using event model scope**, i.e. `@past_events = current_user.attended_events.previous` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/user_events_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations))
* [x] list upcoming events previously attended by user **using event model scope**, i.e. `@upcoming_events = current_user.attended_events.upcoming` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/user_events_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations))

**Review by SpaYco — CHANGES_REQUESTED** [review]
# Changes Requested 

Hello Team! 👋 
Awesome job so far!! 👍 you still need to make some changes though! 

## Required Changes ♻️ 

- [x] on the profile page, `Events you created` should also be separated to upcoming an past ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/user_events_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations))
- [x] add tests: 
  * [x] models have unit tests for validations and associations (Rspec) ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/unit_tests_requirement.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49723079/))
  * [x] authentication have has integration tests (Rspec + Capybara) ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/integration_tests_requirement.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49723079/))
  * [x] event managment features have integration tests (Rspec + Capybara) ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/integration_tests_requirement.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49723079/))
- [x] remove the logic from views [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/section/no_logic_in_helpers_requirement.png) from Pathwright

## OPTIONAL 🟢 

- [x] please use better colors for the design, you can use https://wave.webaim.org/ to help you.

### please make the appropriate changes and ask for another code review ✔️ 

Happy Coding!

Best, 
Aziz Mejri 
_contact me on [twitter](https://twitter.com/iSpaYco) or slack (@SpaYco) if you need any help_

File: app/controllers/users_controller.rb (hunk 1 above)
Comment (dasileker) [reply app/controllers/users_controller.rb]: thanks done with all the requirement .

File: app/controllers/events_controller.rb [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  def show
+    @event = Event.find_by(id: params[:id])
+    @attendees = @event.attendees
+    redirect_to root_url and return unless logged_in?
+  end
+
+  def index
+    @events = Event.all
+    # @previous_events = Event.previous
+    # @upcoming_events = Event.upcoming
```
Comment (cristianCeamatu) [inline app/controllers/events_controller.rb]: [Please fix] Remove commented code.

File: app/controllers/users_controller.rb [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+      render 'new'
+    end
+  end
+
+  def show
+    @user = User.find(params[:id])
+    @organized_events = @user.organized_events
+    @past_events = @user.organized_events.past
+    @upcoming_events = @user.organized_events.upcoming
+    redirect_to root_url and return unless @user == current_user
```
Comment (cristianCeamatu) [inline app/controllers/users_controller.rb:22]: [Optional] Avoid using `and` in statements, `&&` is preferred in ruby.

**Review by cristianCeamatu — CHANGES_REQUESTED** [review]
## ♻️ Changes required

Unfortunately, you did not implement the tests as required:
- [x] models have unit tests for validations and associations (Rspec) ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/unit_tests_requirement.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49723079/))
- [x] authentication have has integration tests (Rspec + Capybara) ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/integration_tests_requirement.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49723079/))
- [x] event managment features have integration tests (Rspec + Capybara) ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/integration_tests_requirement.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49723079/))

This is a very important step because as a developer you need to know how to implement testing. I recommend to slowly follow the tutorials provided by Microverse and implement the Rspec tests because if you learn them one time everything will become much easier later. You will find the tutorials on the project requirements in the Project Specific section (https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49723079/), I would also recommend checking the [shoulda-matchers](https://github.com/thoughtbot/shoulda-matchers) gem which gives some very easy to use matchers.

Please take your time to finish the testing and submit it again. If you need help contact me on slack. Good luck! 💪 

### Please also check my inline comments

Best regards,
Cristian Ceamatu
Slack: Cristian Viorel Ceamatu

File: app/controllers/events_controller.rb (hunk 2 above)
Comment (dasileker) [reply app/controllers/events_controller.rb]: thank you for the review, done  and I hope this is it for all .

File: app/controllers/users_controller.rb (hunk 3 above)
Comment (dasileker) [reply app/controllers/users_controller.rb:22]: done, thank you

**Review by layzrd — APPROVED** [review]
### Status: Approved 🙌 🎇
Hi @dasileker, I am  [Abdulaziz Redi Ali](https://azizali.ml/), your TSE for this PR.  Congratulations 🎊 👏, you did very well on this project.

Now, you can move ahead.

Happy coding 👨‍💻
[Abdulaziz Redi Ali](https://azizali.ml/) | [Contact me on Slack](https://microverse-students.slack.com/team/U011SD6JUH2)

**Review by layzrd — APPROVED** [review]
### Status: Approved 🙌 🎇
Hi @dasileker, I am  [Abdulaziz Redi Ali](https://azizali.ml/), your TSE for this PR.  Congratulations 🎊 👏, you did very well on this project.

Now, you can move ahead.

Happy coding 👨‍💻
[Abdulaziz Redi Ali](https://azizali.ml/) | [Contact me on Slack](https://microverse-students.slack.com/team/U011SD6JUH2)

**Pull request merged**